pyaedt/hfss3dlayout.py

```diff
@@ -645,5 +645,48 @@ def create_frequency_sweep(self, setupname, unit, freqstart, freqstop, num_of_fr
         # self._messenger.add_debug_message("Sweep Setup created correctly")
         return setup
 
+    @aedt_exception_handler
+    def import_gds(self, gds_path, aedb_path=None, xml_path=None, set_as_active=True, close_active_project=False):
+        """Import Gds into HFSS3DLayout and assign stackup from xml if present.
+
+        Parameters
+        ----------
+        gds_path : str
+            Full path to .gds file
+        aedb_path : str, optional
+            Full path to aedb file
+        xml_path : str, optional
+            path to stackup information. If not provided, the stackup will not be edited
+        set_as_active : bool
+            Set Gds as active project
+        close_active_project : bool
+            Close active project after loading the gds.
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        active_project = self.project_name
+        project_name = os.path.basename(gds_path)[:-4]
+        if not aedb_path:
+            aedb_path = gds_path.replace('.gds', '.aedb')
+        if os.path.exists(aedb_path):
+            old_name = project_name
+            project_name = generate_unique_name(project_name)
+            aedb_path = gds_path.replace(old_name + '.gds', project_name + '.aedb')
```

Comment:
```suggestion
            aedb_path = os.path.join(os.path.dirname(gds_path), project_name + '.aedb')
```
It's better to use os.path methods, also, in my opinion, it makes clearer what we do here.